controllers/component_controller_conditions.go

```diff
@@ -141,18 +143,21 @@ func (r *ComponentReconciler) SetGitOpsGeneratedConditionAndUpdateCR(ctx context
 		logutil.LogAPIResourceChangeEvent(log, component.Name, "ComponentGitOpsResources", logutil.ResourceCreate, generateError)
 	}
 	meta.SetStatusCondition(&component.Status.Conditions, condition)
-
 	err := r.Client.Status().Update(ctx, component)
 	if err != nil {
 		// Retry, and if still fails, then return an error
-		curComponent := &appstudiov1alpha1.Component{}
-		err := r.Client.Get(ctx, req.NamespacedName, curComponent)
+		var currentComponent appstudiov1alpha1.Component
+		err := r.Get(ctx, req.NamespacedName, &currentComponent)
 		if err != nil {
-			return err
+			return nil
```

Comment:
same comment as before

Reply:
updated to return error if fail to get component at the retry